File: src/node/magma/app/compile/type/BooleanType.ts
import { Type } from "../../../../magma/api/Type";
import { Platform } from "../../../../magma/app/io/Platform";
export class BooleanType implements Type {
	platform: Platform;
	constructor (platform: Platform) {
		this.platform = platform;
	}
	generate(): string {
		if (Platform/*auto*/.TypeScript === this/*auto*/.platform) {
			return "boolean";
		}
		return "Bool";
	}
	isFunctional(): boolean {
		return false/*auto*/;
	}
	isVar(): boolean {
		return false/*auto*/;
	}
	generateBeforeName(): string {
		return "";
	}
}

File: src/java/magma/app/compile/type/BooleanType.java
package magma.app.compile.type;

import magma.api.Type;
import magma.app.io.Platform;

public record BooleanType(Platform platform) implements Type {
    @Override
    public String generate() {
        if (Platform.TypeScript == this.platform) {
            return "boolean";
        }

        return "Bool";
    }

    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }
}
